kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -270,6 +296,20 @@ private AtomicWriteResult flush(final long offset, final UUID txnid) {
 
   @Override
   public void restoreBatch(final Collection<ConsumerRecord<byte[], byte[]>> records) {
+    final List<ConsumerRecord<byte[], byte[]>> batch = new LinkedList<>();
```

Comment:
we should definitely use an `ArrayList` here and just size it to the `maxBatchSize`

I've had some really surprisingly bad experiences with LinkedLists where something innocuous like this causes really heavy GC load

Reply:
+1

Reply:
Since the max batch size is a constant, we should just create the ArrayList of that size up front in the constructor and then we can reuse it here. Especially since we're repeatedly clearing/refilling it anyway

Reply:
Yep, good call on the ArrayList. Using a long-lived buffer gives me the heebie-jeebies a little bit. I've been bitten in the past by that kind of thing where something winds up sticking around in the buffer that doesn't get cleared between calls for whatever reason (though of course we could/should clear it as the first thing the method does, it's somewhat hard to guarantee that always happens as the class evolves). I don't think there is a significant cost to just allocating it on each restore batch.